site/content/docs/developing/storage.md

```diff
@@ -0,0 +1,219 @@
+# Storage
+
+There are two ways to add persistence to Copilot workloads: using [`copilot storage init`](#database-and-artifacts) to create databases and S3 buckets; and attaching an existing EFS filesystem using the [`storage` key](#file-systems) in the manifest. 
+
+## Database and Artifacts
+
+To add a database or S3 bucket to your job or service, simply run [`copilot storage init`](../commands/storage-init.md).
+```bash
+$ copilot storage init -n my-bucket -t S3 -w api
+```
+The above command will create the Cloudformation template for an S3 bucket in the [addons](../developing/additional-aws-resources.md) directory for the "api" service. The next time you run `copilot deploy -n api`, the bucket will be created, permission to access it will be added to the `api` task role, and the name of the bucket will be injected into the `api` container under the environment variable `MY_BUCKET_NAME`. 
+
+All names are converted into SCREAMING_SNAKE_CASE based on their use of hyphens or underscores. You can view the environment variables for a given service by running `copilot svc show`.
+
+You can also create a [DynamoDB table](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Introduction.html) using `copilot storage init`. For example, to create the Cloudformation template for a table with a sort key and a local secondary index, you could run the following command.
+
+```bash
+$ copilot storage init -n users -t DynamoDB -w api --partition-key id:N --sort-key email:S --lsi post-count:N
```

Comment:
I wonder if we should not show all these flags? To encourage users to use the guided path instead.

I saw in many of our issues that folks use the flags instead of prompts. I wonder if it's because of our examples or docs.

Reply:
Or maybe we can write it like this:
```suggestion
# For a guided experience.
$ copilot storage init

# Or by providing the flags to skip the prompts.
$ copilot storage init -n users -t DynamoDB -w api --partition-key id:N --sort-key email:S --lsi post-count:N
```